cypress/views/mcg-ms-common.ts

```diff
@@ -0,0 +1,22 @@
+import { DATA_FEDERATION, DATA_SERVICES, NO_WAIT } from '../constants/common';
+
+export const MCG_MS_Common = {
+  visitMcgMsDashboard: () => {
+    cy.get('[data-test="nav"]', { timeout: 15 * 1000 })
+      .should('contain', DATA_SERVICES)
+      .within(() => {
+        cy.contains(DATA_SERVICES, NO_WAIT)
+          .should('have.attr', 'aria-expanded', 'false')
+          .click();
+        cy.contains(DATA_SERVICES, NO_WAIT).should(
+          'have.attr',
+          'aria-expanded',
+          'true'
+        );
+        cy.contains(DATA_FEDERATION, NO_WAIT)
+          .should('have.attr', 'href', '/mcgms/cluster')
+          .click();
+      });
+    cy.location('pathname', NO_WAIT).should('eq', '/mcgms/cluster');
+  },
+};
```

Comment:
clickNavLink is not working and took this much long to figure out

Reply:
here this [line](https://github.com/debjyoti-pandit/mcg-ms-console/blob/f0224f026d85c92d9f94618a2b25716ce0988dfb/cypress/support/selectors.ts#L83) 

retries only the last command.
since our data services get attach a second or 2 late it is failing

Reply:
ref: https://docs.cypress.io/guides/core-concepts/retry-ability#Only-the-last-command-is-retried

Reply:
it is working for the storage class in the OBC test as Storage gets loaded as soon as the main screen is loaded.
but our data federation is attached a few seconds later.

Reply:
then fix clickNavLink so that it can be reused... instead of adding here... we can not keep on adding things for specific use cases...

Reply:
so, it is waiting https://github.com/debjyoti-pandit/mcg-ms-console/blob/f0224f026d85c92d9f94618a2b25716ce0988dfb/cypress/support/selectors.ts#L82 for Data services for about 10s... u r saying that is not enough ??

Reply:
> so, it is waiting https://github.com/debjyoti-pandit/mcg-ms-console/blob/f0224f026d85c92d9f94618a2b25716ce0988dfb/cypress/support/selectors.ts#L82 for Data services for about 10s... u r saying that is not enough ??

no it will not work in this case.
not because of time but because of the yield of `byTestID`

`byTestID` will never have the data services as when it is loaded there were no data services and it is never retried.
only the content is retried

Reply:
I agree... `byTestID` on L81 will not have data services, but do not need it as well... cause it is only checking for "nav"...
once "nav" is found then it will move to "contains" that will check for data services...

Reply:
so as a solution u have increased { timeout: 15 * 1000 }, why not do it directly in `clickNavLink`...

Reply:
it's not the time, we have to then squash the 2 command in 1 so it retries the complete action

Reply:
let;s connect over a call to discuss... if this is going to continue like this...

Reply:
i can still see tests failing at this point...

Reply:
> let;s connect over a call to discuss... if this is going to continue like this...

let we know when